### PR TITLE
merge-from-heroku-upstream

### PR DESCRIPTION
**Pull request by jhwillett**
PR for https://github.com/ProsperWorks/heroku-buildpack-ruby, pulls in a half-year of updates there.

### CODE REVIEW

**Comment by jhwillett** [comment]
The .travis.yml depends on some secrets which include a HEROKU_API_KEY.

I have disabled TravisCI and the TravisCI requirements for `master` on this repo, but unfortunately those big red Xs still show up.

Please disregard: but I acknowledge it is icky that we don't have CI for this.

File: circle.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ machine:
 
 test:
   pre:
+    - wget -qO- https://cli-assets.heroku.com/install-ubuntu.sh | sh
```
Comment (marshall-prosperworks) [inline circle.yml:12]: Lovely

**Review by marshall-prosperworks — APPROVED** [review]

File: circle.yml (hunk 1 above)
Comment (jhwillett) [reply circle.yml:12]: Haha

Comment (jhwillett) [reply circle.yml:12]: Kind of intense that they have a `circle.yml` and `.travis.yml` both.

**Pull request merged**